### PR TITLE
Adding support for generic controller

### LINKED ISSUE #47: Adding support for generic controller
**Issue/Feature Description:**
Adding support for generic controller
**Why this issue to fixed / feature is needed(give scenarios or use cases):**
Support generic controller
**How to reproduce, in case of a bug:**
NA
**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**
NA

### PR DESCRIPTION
**Pull request by AmitRoushan**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

/kind enhancement

**What this PR does / why we need it**:

Adding support for generic controller

**Which issue(s) this PR fixes**:

Fixes #47 

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED

/kind TESTED

**Test Report**:

NA 
**Special notes for your reviewer**:

### CODE REVIEW

**Review by sushanthakumar — DISMISSED** [review]

File: controllers/app/app.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	kahuClient, err := clientFactory.KahuClient()
+	if err != nil {
+		return fmt.Errorf("failed to initialize kahu client %s", err)
+	}
+
+	// initialize shared informer framework
+	sharedInformerFactory := informers.NewSharedInformerFactoryWithOptions(kahuClient, 0)
+
+	// initialize all controllers
+	ctrl := restore.NewRestoreController(sharedInformerFactory.Kahu().V1beta1().Restores())
```
[the remark this answers is not in the document]
Comment (AmitRoushan) [reply controllers/app/app.go]: The code has been removed to handle controller statup framework

File: controllers/app/app.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
```
[the remark this answers is not in the document]
Comment (AmitRoushan) [reply controllers/app/app.go]: The code has been removed to handle controller statup framework. The comment is handled in the framework

File: controllers/app/app.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	// add required controllers to manager
+	mgr.Add(manager.RunnableFunc(func(ctx context.Context) error {
```
[the remark this answers is not in the document]
Comment (AmitRoushan) [reply controllers/app/app.go]: The code has been removed to handle controller statup framework

**Review by sushanthakumar — APPROVED** [review]

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Pull request merged**